statistics/cmsketch.go

```diff
@@ -22,17 +23,30 @@ import (
 	"github.com/pingcap/tidb/sessionctx/stmtctx"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/hack"
 	"github.com/pingcap/tipb/go-tipb"
 	"github.com/spaolacci/murmur3"
 )
 
+// TopNThreshold is the minimum ratio of the number of topn elements in CMSketch, 20 means 1 / 20 = 5%.
+const TopNThreshold = 20
+
 // CMSketch is used to estimate point queries.
 // Refer: https://en.wikipedia.org/wiki/Count-min_sketch
+// If topnlimit is 0, topn will be readonly
+// Once constructed with topn, no new values should be inserted, or statistic data will be wrong.
 type CMSketch struct {
-	depth int32
-	width int32
-	count uint64
-	table [][]uint32
+	depth     int32
+	width     int32
+	count     uint64
+	table     [][]uint32
+	topn      []cmscount
```

Comment:
This field `topn` can be removed?
We can just let it as a function scope variable in `BuildTopN`?